roles/matrix-client-hydrogen/tasks/setup.yml

```diff
@@ -0,0 +1,108 @@
+---
+
+#
+# Tasks related to setting up Hydrogen
+#
+
+- name: Ensure Hydrogen paths exists
+  file:
+    path: "{{ item.path }}"
+    state: directory
+    mode: 0750
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+  with_items:
+    - { path: "{{ matrix_client_hydrogen_data_path }}", when: true }
+    - { path: "{{ matrix_client_hydrogen_docker_src_files_path }}", when: "{{ matrix_client_hydrogen_container_image_self_build }}" }
+  when: matrix_client_hydrogen_enabled|bool and item.when
+
+- name: Ensure Hydrogen Docker image is pulled
+  docker_image:
+   name: "{{ matrix_client_hydrogen_docker_image }}"
+   source: "{{ 'pull' if ansible_version.major > 2 or ansible_version.minor > 7 else omit }}"
+   force_source: "{{ matrix_client_hydrogen_docker_image_force_pull if ansible_version.major > 2 or ansible_version.minor >= 8 else omit }}"
+   force: "{{ omit if ansible_version.major > 2 or ansible_version.minor >= 8 else matrix_client_hydrogen_docker_image_force_pull }}"
+  when: matrix_client_hydrogen_enabled|bool and not matrix_client_hydrogen_container_image_self_build
+
+- name: Ensure Hydrogen repository is present on self-build
+  git:
+    repo: "{{ matrix_client_hydrogen_container_image_self_build_repo }}"
+    dest: "{{ matrix_client_hydrogen_docker_src_files_path }}"
+    version: "{{ matrix_client_hydrogen_docker_image.split(':')[1] }}"
+    force: "yes"
+  register: matrix_client_hydrogen_git_pull_results
+  when: "matrix_client_hydrogen_enabled|bool and matrix_client_hydrogen_container_image_self_build|bool"
+
+- name: Ensure Hydrogen configuration installed
+  copy:
+    content: "{{ matrix_client_hydrogen_configuration|to_nice_json }}"
+    dest: "{{ matrix_client_hydrogen_docker_src_files_path }}/assets/config.json"
```

Comment:
Shouldn't this be:

```yaml
dest: "{{ matrix_client_hydrogen_data_path }}/config.json"
```

Reply:
No. I tried that first but Hydrogen only looks for a `config.json` when it gets built. It doesn't work like Element. https://github.com/vector-im/hydrogen-web/issues/324

Reply:
Ah, I see. That's kind of misleading then.

So, in the general case (no self-building), we'll use the [default upstream configuration](https://github.com/vector-im/hydrogen-web/blob/master/assets/config.json), which points us to `matrix.org`. That's not really ideal and will hopefully be addressed in vector-im/hydrogen-web#324 (or some other issue), so that users would just need to enter their localpart (`.well-known` may be made to work in the future, but that's asking people to enter their full MXID is not ideal).

Reply:
Yeah. At the moment the pre-built images no not work so self building will be enabled by default.